arch/arm/core/aarch64/switch.S

```diff
@@ -152,7 +133,11 @@ SECTION_FUNC(TEXT, z_thread_entry_wrapper)
 
 GTEXT(z_arm64_svc)
 SECTION_FUNC(TEXT, z_arm64_svc)
-	z_arm64_enter_exc x0, x1, x2
+	z_arm64_enter_exc x2, x3, x4
+
+	/* Preserve the parameters passed from z_arm64_call_svc() */
+	mov	x10, x0
```

Comment:
Please use callee saved ones.. x19 x20
offload path can clobber. Even otherwise also its safer practice.

Reply:
We cannot use any callee saved register here because in case of context switch those are going to be saved later in the thread context (see `z_arm64_context_switch()`) so we are forced to use one of the register that we have already saved as part of `z_arm64_enter_exc()`. The optimal solution here would be to save those onto the stack but the code would end up be way more complicated, so I just decided to use some temporary register to hold the `x0` and `x1` values.

Reply:
About the offload path I don't think we care about that. `z_irq_do_offload` can clobber `x10` and `x11` but in the exit path we are going to restore those anyway in `z_arm64_enter_exc()`.